_pytest/config.py

```diff
@@ -99,7 +99,7 @@ def directory_arg(path, optname):
      "mark main terminal runner python fixtures debugging unittest capture skipping "
      "tmpdir monkeypatch recwarn pastebin helpconfig nose assertion "
      "junitxml resultlog doctest cacheprovider freeze_support "
-     "setuponly setupplan warnings").split()
+     "setuponly setupplan").split()
```

Comment:
> the most apparent blunder being the addition of the once filter

Not really, unfortunately `catch_warnings` will remove **all** current filters in order to be able to catch all warnings. That filter is there just to prevent the same warning to be triggered more than once by the same test.

How would you like to make this change without making a 4.0 release? Or do you want to make a 4.0 release regardless?